### ISSUE TITLE
Extremely slow, what am I doing wrong? wan2.1 t2v

### ISSUE THREAD

**Issue by tralalog** [kohya-ss/musubi-tuner#489]
People on reddit with similar computer specs are saying they are training loras in around 5 hours.  Why are my trainings coming to 50-100 hours?  I am training Wan2_1-T2V-14B_fp8_e4m3fn. I just did a fresh install with the same results.  I am on win 10 running ubuntu 22.04 lts.

My rig:
i5-9600k (with vmx on), 64 gb ddr4 3200, 1tb samsung ssd, rtx 5060 ti 16gb. i am training on 13 images.

my toml:

[general]
resolution = [340, 512]
enable_bucket = true
bucket_no_upscale = false
caption_extension = ".txt"
batch_size = 1

[[datasets]]
image_directory = "/home/joe/musubi-tuner/dataset/images"
cache_directory = "/home/joe/musubi-tuner/dataset/cache"
num_repeats = 1
fp_latent_window_size = 1

my training:

python src/musubi_tuner/wan_cache_latents.py --dataset_config dataset/toml/dataset_config.toml --vae models/wan/Wan2.1_VAE.pth

python src/musubi_tuner/wan_cache_text_encoder_outputs.py --dataset_config dataset/toml/dataset_config.toml  --t5 models/wan/models_t5_umt5-xxl-enc-bf16.pth --batch_size 1

accelerate launch --num_cpu_threads_per_process 1 \
  src/musubi_tuner/wan_train_network.py \
  --task t2v-14B \
  --dit ./models/wan/Wan2_1-T2V-14B_fp8_e4m3fn.safetensors \
  --dataset_config ./dataset/toml/dataset_config.toml \
  --mixed_precision bf16 --sdpa --fp8_base \
  --optimizer_type adamw8bit --learning_rate 3e-4 \
  --gradient_accumulation_steps 1 \
  --max_data_loader_n_workers 6 --persistent_data_loader_workers \
  --network_module networks.lora_wan --network_dim 16 \
  --max_train_steps 1300 --save_every_n_steps 100 \
  --seed 42 --optimizer_args weight_decay=0.1 --max_grad_norm 0 \
  --lr_scheduler polynomial --lr_scheduler_power 4 --lr_scheduler_min_lr_ratio 5e-5 \
  --dynamo_backend NO \
  --output_dir ./dataset/output --output_name n4ncyv1

some lines in there from chatgpt trying to help. ive tried change the network/alpha and see no impact.

**Comment by kohya-ss** [comment]
Please add `--gradient_checkpointing` option. This option significantly reduces VRAM usage.

**Comment by tralalog** [comment]
--gradient_checkpointing makes no difference

**Comment by niceguy4** [comment]
Can you run this command within your virtual environment and copy and paste here.

`(venv) D:\musubi-tuner>pip freeze show`

Also, copy and paste from the point you enter your accelerate launch command to the 1st epoch. 

Also, did you do the `accelerate config` ?

**Comment by tralalog** [comment]
i really appreciate the help.

(musubi-tuner) joe@DESKTOP-Q4R8MGK:~$ pip freeze show
accelerate==1.6.0
av==14.0.1
bitsandbytes==0.45.5
certifi==2025.8.3
charset-normalizer==3.4.3
diffusers==0.32.1
easydict==1.13
einops==0.7.0
filelock==3.19.1
fsspec==2025.7.0
ftfy==6.3.1
hf-xet==1.1.8
huggingface-hub==0.34.3
idna==3.10
importlib_metadata==8.7.0
Jinja2==3.1.6
MarkupSafe==3.0.2
mpmath==1.3.0
-e git+https://github.com/kohya-ss/musubi-tuner@e96be1caae55ca6dc95aefcc1018d10b9090dd9f#egg=musubi_tuner
networkx==3.5
numpy==2.3.2
nvidia-cublas-cu12==12.8.4.1
nvidia-cuda-cupti-cu12==12.8.90
nvidia-cuda-nvrtc-cu12==12.8.93
nvidia-cuda-runtime-cu12==12.8.90
nvidia-cudnn-cu12==9.10.2.21
nvidia-cufft-cu12==11.3.3.83
nvidia-cufile-cu12==1.13.1.3
nvidia-curand-cu12==10.3.9.90
nvidia-cusolver-cu12==11.7.3.90
nvidia-cusparse-cu12==12.5.8.93
nvidia-cusparselt-cu12==0.7.1
nvidia-nccl-cu12==2.27.5
nvidia-nvjitlink-cu12==12.8.93
nvidia-nvshmem-cu12==3.3.20
nvidia-nvtx-cu12==12.8.90
opencv-python==4.10.0.84
packaging==25.0
pillow==11.3.0
psutil==7.0.0
pytorch-triton==3.4.0+gitf7888497
PyYAML==6.0.2
regex==2025.7.34
requests==2.32.5
safetensors==0.4.5
sageattention==1.0.6
sentencepiece==0.2.0
setuptools==78.1.1
sympy==1.14.0
tokenizers==0.21.4
toml==0.10.2
torch==2.9.0.dev20250822+cu128
torchaudio==2.8.0.dev20250822+cu128
torchvision==0.24.0.dev20250822+cu128
tqdm==4.67.1
transformers==4.54.1
typing_extensions==4.14.1
urllib3==2.5.0
voluptuous==0.15.2
wcwidth==0.2.13
wheel==0.45.1
zipp==3.23.0

accelerate config:
This machine
No distributed training
cpu only - NO
torch dynamo - NO
deepseed - NO
gpu - all (my monitor is on onboard apu, so training gets all of gpu)
numa effeciency - NO
precision - bf16

accelerate launch --num_cpu_threads_per_process 1 \
src/musubi_tuner/wan_train_network.py \
  --task t2v-14B \
  --dit ./models/wan/Wan2_1-T2V-14B_fp8_e4m3fn.safetensors \
  --dataset_config ./dataset/toml/dataset_config.toml \
  --mixed_precision bf16 --sdpa --fp8_base \
  --optimizer_type adamw8bit --learning_rate 3e-4 \
  --gradient_checkpointing \
  --gradient_accumulation_steps 1 \
  --max_data_loader_n_workers 6 --persistent_data_loader_workers \
  --network_module networks.lora_wan --network_dim 16 \
  --max_train_steps 1300 --save_every_n_steps 100 \
  --seed 42 --optimizer_args weight_decay=0.1 --max_grad_norm 0 \
  --lr_scheduler polynomial --lr_scheduler_power 4 --lr_scheduler_min_lr_ratio 5e-5 \
  --dynamo_backend NO \
  --output_dir ./dataset/output --output_name n4ncyv1

----

(musubi-tuner) joe@DESKTOP-Q4R8MGK:~/musubi-tuner$ accelerate launch --num_cpu_threads_per_process 1 \
> src/musubi_tuner/wan_train_network.py \
>   --task t2v-14B \
>   --dit ./models/wan/Wan2_1-T2V-14B_fp8_e4m3fn.safetensors \
>   --dataset_config ./dataset/toml/dataset_config.toml \
>   --mixed_precision bf16 --sdpa --fp8_base \
optimize>   --optimizer_type adamw8bit --learning_rate 3e-4 \
adient_>   --gradient_checkpointing \
  --grad>   --gradient_accumulation_steps 1 \
>   --max_data_loader_n_workers 6 --persistent_data_loader_workers \
>   --network_module networks.lora_wan --network_dim 16 \
ax_trai>   --max_train_steps 1300 --save_every_n_steps 100 \
seed 42 >   --seed 42 --optimizer_args weight_decay=0.1 --max_grad_norm 0 \
_schedu>   --lr_scheduler polynomial --lr_scheduler_power 4 --lr_scheduler_min_lr_ratio 5e-5 \
 --dynam>   --dynamo_backend NO \
>   --output_dir ./dataset/output --output_name n4ncyv1
Trying to import sageattention
Successfully imported sageattention
INFO:musubi_tuner.wan.modules.model:Detected DiT dtype: torch.float8_e4m3fn
INFO:musubi_tuner.hv_train_network:Load dataset config from dataset/toml/dataset_config.toml
INFO:musubi_tuner.dataset.image_video_dataset:glob images in /home/joe/musubi-tuner/dataset/images
INFO:musubi_tuner.dataset.image_video_dataset:found 13 images
INFO:musubi_tuner.dataset.config_utils:[Dataset 0]
  is_image_dataset: True
  resolution: (340, 512)
  batch_size: 1
  num_repeats: 1
  caption_extension: ".txt"
  enable_bucket: True
  bucket_no_upscale: False
  cache_directory: "/home/joe/musubi-tuner/dataset/cache"
  debug_dataset: False
    image_directory: "/home/joe/musubi-tuner/dataset/images"
    image_jsonl_file: "None"
    fp_latent_window_size: 1
    fp_1f_clean_indices: None
    fp_1f_target_index: None
    fp_1f_no_post: False
    flux_kontext_no_resize_control: False
    qwen_image_edit_no_resize_control: False
    qwen_image_edit_control_resolution: None


INFO:musubi_tuner.dataset.image_video_dataset:bucket: (336, 512, 1), count: 8
INFO:musubi_tuner.dataset.image_video_dataset:bucket: (512, 336, 1), count: 5
INFO:musubi_tuner.dataset.image_video_dataset:total batches: 13
INFO:musubi_tuner.hv_train_network:preparing accelerator
accelerator device: cuda
INFO:musubi_tuner.hv_train_network:DiT precision: torch.bfloat16, weight precision: torch.float8_e4m3fn
INFO:musubi_tuner.hv_train_network:Loading DiT model from ./models/wan/Wan2_1-T2V-14B_fp8_e4m3fn.safetensors
INFO:musubi_tuner.wan.modules.model:Creating WanModel. I2V: False, FLF2V: False, V2.2: False, device: cuda, loading_device: cuda, fp8_scaled: False
INFO:musubi_tuner.wan.modules.model:Loading DiT model from ./models/wan/Wan2_1-T2V-14B_fp8_e4m3fn.safetensors, device=cuda
INFO:musubi_tuner.utils.lora_utils:Loading model files: ['./models/wan/Wan2_1-T2V-14B_fp8_e4m3fn.safetensors']
INFO:musubi_tuner.utils.lora_utils:Loading state dict without FP8 optimization. Dtype of weight: None, hook enabled: False
INFO:musubi_tuner.wan.modules.model:Casting model weights to torch.float8_e4m3fn
INFO:musubi_tuner.wan.modules.model:Loaded DiT model from ./models/wan/Wan2_1-T2V-14B_fp8_e4m3fn.safetensors, info=<All keys matched successfully>import network module: networks.lora_wan
INFO:musubi_tuner.networks.lora:create LoRA network. base dim (rank): 16, alpha: 1
INFO:musubi_tuner.networks.lora:neuron dropout: p=None, rank dropout: p=None, module dropout: p=None
INFO:musubi_tuner.networks.lora:create LoRA for U-Net/DiT: 400 modules.
INFO:musubi_tuner.networks.lora:enable LoRA for U-Net: 400 modules
WanModel: Gradient checkpointing enabled.
prepare optimizer, data loader etc.
INFO:musubi_tuner.hv_train_network:use 8-bit AdamW optimizer | {'weight_decay': 0.1}
INFO:musubi_tuner.hv_train_network:casting model to torch.float8_e4m3fn
running training / 学習開始
  num train items / 学習画像、動画数: 13
  num batches per epoch / 1epochのバッチ数: 13
  num epochs / epoch数: 100
  batch size per device / バッチサイズ: 1
  gradient accumulation steps / 勾配を合計するステップ数 = 1
  total optimization steps / 学習ステップ数: 1300
INFO:musubi_tuner.hv_train_network:set DiT model name for metadata: ./models/wan/Wan2_1-T2V-14B_fp8_e4m3fn.safetensors
steps:   0%|                                                                                                             | 0/1300 [00:00<?, ?it/s]INFO:musubi_tuner.hv_train_network:DiT dtype: torch.float8_e4m3fn, device: cuda:0

epoch 1/100
INFO:musubi_tuner.dataset.image_video_dataset:epoch is incremented. current_epoch: 0, epoch: 1
INFO:musubi_tuner.dataset.image_video_dataset:epoch is incremented. current_epoch: 0, epoch: 1
INFO:musubi_tuner.dataset.image_video_dataset:epoch is incremented. current_epoch: 0, epoch: 1
INFO:musubi_tuner.dataset.image_video_dataset:epoch is incremented. current_epoch: 0, epoch: 1
INFO:musubi_tuner.dataset.image_video_dataset:epoch is incremented. current_epoch: 0, epoch: 1
INFO:musubi_tuner.dataset.image_video_dataset:epoch is incremented. current_epoch: 0, epoch: 1
steps:   1%|▊                                                                               | 13/1300 [16:05<26:32:51, 74.26s/it, avr_loss=0.0622]
epoch 2/100

**Comment by kohya-ss** [comment]
The configuration seems fine. There may be an issue with the PyTorch version or the library's support for the RTX 50 series, but I don't have an RTX 50 series GPU so I can't verify it.

**Comment by niceguy4** [comment]
I wonder if the bf16 is causing issues. Try this on the setup.

`--mixed_precision fp16`

On the accelerate config during the setup, _precision_ (no or none) whatever the option issues. 

Also, you are on the nightly build for torch. This may or may not be causing issue. 

`pip uninstall torch torchvision`

`pip install torch torchvision --index-url https://download.pytorch.org/whl/cu129`

**Comment by Sarania** [comment]
Yeah you should be going SO much faster than that, I have i7 6900k + 64GB DDR4 + 4070 Ti Super 16GB so not that different specs and I could train the LoRA you're trying to train in an hour or two at most even without my extra optimizations. I do see you're not using ANY blocks to swap so I wonder if you are getting into shared VRAM which will slow down things massively. Even in pure fp8_e4m3fn the base Wan 2.1 DiT is 15GB then plus your dataset, optimizer, gradients, LoRA there's no way you aren't over even at a fairly low res.  I don't have a Windows install to show what I'm talking about but if you look under task manager, GPU section you should see a section for shared memory - if that's filling up, that might be your problem. In any case, my recommendation would be to add `--blocks_to_swap 10` and see if that makes a difference. Also as @niceguy4 mentioned I would change to `--mixed_precision fp16` I don't think that's your problem but it should be better for Wan unless you are using the bf16 base which you're not. Hopefully that can help, good luck!

**Comment by tralalog** [comment]
well it seems it may have been because I was running musubi in wsl2 ubuntu. i installed the windows musubi tonight and its going good.  this is what I am using if it helps anyone else:

pip install torch==2.7.1 torchvision==0.22.1 torchaudio==2.7.1 --index-url https://download.pytorch.org/whl/cu128

**Comment by Sarania** [comment]
WSL2 shouldn't really be slower but regardless I'm glad you found success!
